spras/omicsintegrator2.py

```diff
@@ -148,14 +148,20 @@ def parse_output(raw_pathway_file, standardized_pathway_file):
         """
         # Omicsintegrator2 returns a single line file if no network is found
         num_lines = sum(1 for line in open(raw_pathway_file))
+        # Omicsintegrator2 has corrupted output; list of correct column names
+        sorted_correct_column_names = ['cost', 'in_solution', 'protein1', 'protein2']
+
         if num_lines < 2:
             df = pd.DataFrame(columns=['Node1', 'Node2', 'Rank', 'Direction'])
         else:
             df = pd.read_csv(raw_pathway_file, sep='\t', header=0)
-            df = df[df['in_solution'] == True]  # Check whether this column can be empty before revising this line
-            df = df.take([0, 1], axis=1)
-            df = add_rank_column(df)
-            df = reinsert_direction_col_undirected(df)
-            df.columns = ['Node1', 'Node2', 'Rank', "Direction"]
+            if sorted(df.columns) == sorted_correct_column_names: # if column header names are all correct
+                df = df[df['in_solution'] == True]  # Check whether this column can be empty before revising this line
+                df = df.take([0, 1], axis=1)
```

Comment:
This will have to change now that the columns are in a new order. We should use the column names anyway instead of `take`, which is less readable.

Reply:
In our meeting discussion we agreed this approach works. However, we may want a comment about what `take` on the first two column is the right thing to do. We can also update the column about what the above line is doing now that we've studied `in_solution` more and know what it is doing.